rag/app/test_semantic_search.py

```diff
@@ -0,0 +1,66 @@
+from dotenv import load_dotenv
+import os
+import psycopg2
+import openai
+
+load_dotenv()
+OPENAI_API_KEY = os.environ["OPENAI_API_KEY"]
+openai.api_key = OPENAI_API_KEY
+POSTGRES_USER = os.environ["POSTGRES_USER"]
+POSTGRES_PASSWORD = os.environ["POSTGRES_PASSWORD"]
+POSTGRES_DB = os.environ["POSTGRES_DB"]
+POSTGRES_HOST = "localhost"
+POSTGRES_PORT = "5432"
+
+# Database connection parameters
+DB_PARAMS = {
+    "user": POSTGRES_USER,
+    "password": POSTGRES_PASSWORD,
+    "database": POSTGRES_DB,
+    "host": POSTGRES_HOST,
+    "port": POSTGRES_PORT,
+}
+
```

Comment:
do we have to store repetitive code in an external file?

Reply:
Refactored this with /config.py at root which stores all env variables. Had to refactor the docker-compose.yml as well as individual service dockerfiles. Implemented in 44697b00916c95a35fc0c14b7d669a8005af9d73 and 065d019a38589e924a0c507f545fe5e1461aa94a.